node/v2/handler.js

```diff
@@ -90,6 +89,24 @@ TChannelV2Handler.prototype._write = function _write(frame, encoding, callback)
     }
 };
 
+TChannelV2Handler.prototype._read = function _read(/* n */) {
+    /* noop */
+};
+
+TChannelV2Handler.prototype.sendFrame = function(frame, callback) {
+    var self = this;
+    if (!callback) {
+        callback = function emitIt(err) {
+            if (err) self.emit('error', err);
+        };
+    }
+    if (self.push(frame)) {
+        callback();
+    } else {
+        self.once('drain', callback);
```

Comment:
you are pushing into your readable side and waiting for your writable side to drain. That's not how it works.

if `write()` returns `false` you should listen to `'drain'`.

I don't know what to listen for if `push()` returns `false`. Maybe `readable` ?